lasagne/layers/recurrent.py

```diff
@@ -757,3 +762,306 @@ def step_masked(input_dot_W_n, mask_n, cell_previous, hid_previous,
+                 W_hid_to_cell=init.Normal(0.1),
+                 b_cell=init.Constant(0.),
+                 nonlinearity_resetgate=nonlinearities.sigmoid,
+                 nonlinearity_updategate=nonlinearities.sigmoid,
+                 nonlinearity_cell=nonlinearities.tanh,
+                 hid_init=init.Constant(0.),
+                 learn_init=True,
+                 backwards=False,
+                 gradient_steps=-1,
+                 grad_clipping=False,
+                 unroll_scan=False):
+
+        # Initialize parent layer
+        super(GRULayer, self).__init__(incoming)
+        # For any of the nonlinearities, if None is supplied, use identity
+        if nonlinearity_resetgate is None:
+            self.nonlinearity_resetgate = nonlinearities.identity
+        else:
+            self.nonlinearity_resetgate = nonlinearity_resetgate
+
+        if nonlinearity_updategate is None:
+            self.nonlinearity_updategate = nonlinearities.identity
+        else:
+            self.nonlinearity_updategate = nonlinearity_updategate
+
+        if nonlinearity_cell is None:
+            self.nonlinearity_cell = nonlinearities.identity
+        else:
+            self.nonlinearity_cell = nonlinearity_cell
+
+        self.learn_init = learn_init
+        self.num_units = num_units
+        self.grad_clipping = grad_clipping
+        self.backwards = backwards
+        self.gradient_steps = gradient_steps
+        self.unroll_scan = unroll_scan
+
+        # Input dimensionality is the output dimensionality of the input layer
+        num_batch = self.input_shape[0]
+        num_inputs = np.prod(self.input_shape[2:])
+        self.num_inputs = num_inputs
+
+        self.W_in_to_updategate = self.add_param(
+            W_in_to_updategate, (num_inputs, num_units),
+            name="W_in_to_updategate")
+
+        self.W_hid_to_updategate = self.add_param(
+            W_hid_to_updategate, (num_units, num_units),
+            name="W_hid_to_updategate")
+
+        self.b_updategate = self.add_param(
+            b_updategate, (num_units,),
+            name="b_updategate", regularizable=False)
+
+        self.W_in_to_resetgate = self.add_param(
+            W_in_to_resetgate, (num_inputs, num_units),
+            name="W_in_to_resetgate")
+
+        self.W_hid_to_resetgate = self.add_param(
+            W_hid_to_resetgate, (num_units, num_units),
+            name="W_hid_to_resetgate")
+
+        self.b_resetgate = self.add_param(
+            b_resetgate, (num_units,),
+            name="b_resetgate", regularizable=False)
+
+        self.W_in_to_cell = self.add_param(
+            W_in_to_cell, (num_inputs, num_units), name="W_in_to_cell")
+
+        self.W_hid_to_cell = self.add_param(
+            W_hid_to_cell, (num_units, num_units), name="W_hid_to_cell")
+
+        self.b_cell = self.add_param(
+            b_cell, (num_units,), name="b_cell", regularizable=False)
+
+        self.W_in_to_gates = T.concatenate(
```

Comment:
`to_gates` doesn't make a lot of sense for `W_in_to_cell`.  I would be fine calling this simply `W_in` (and `W_hid` below) or `W_in_stacked` or something.  This is true for all of the layers, so you can ignore it here...